### DISCUSSION TITLE
About the recent changes of property naming

### DISCUSSION THREAD

**Discussion by diskerr** [android/nowinandroid#1124] [General]
PR #1094 (merged)  has left me a bit confused.

In non-Compose code, changing the naming of constants to `CAPITALS_AND_UNDERSCORES` seems reasonable to follow the official Kotlin style guide, although two distinct styles for constant naming coexist within a single codebase.

However, in the Compose code, the official [Style guidelines for Jetpack Compose APIs](https://developer.android.com/jetpack/compose/api-guidelines) supersedes this and clearly states that the constants in the Compose framework **MUST** use `PascalCase`. In addition, it indicates Compose libraries **SHOULD** follow this, and apps **MAY** do so. Thus, I assumed that the NiA app also followed the PascalCase style, in accordance with this **MAY** rule, until I saw the above-mentioned PR.

For example, [in this change](https://github.com/android/nowinandroid/pull/1094/files#diff-f06d8bc3fe154a604e0b96f79fd2ef64e6e5155479338c5e6b670bf6ff6a0955L127), the property `const val DisabledChipContainerAlpha: Float` is a constant. Thus, imho, using `PascalCase` for the property name is **not a violation** of the above style guidelines, and changing it to `DISABLED_CHIP_CONTAINER_ALPHA` might lead to confusion for readers. One might wonder why `NiaIconButtonDefaults.CONSTANT_PROPERTY` follows `CAPITALS_AND_UNDERSCORES`, while `ButtonDefaults.ConstantProperty` in Compose uses `PascalCase`, and confusing "Which one should I follow?"

Why should we fix the style?

**Comment by lihenggui** [comment]
This is because Ktlint disallows the use of PascalCase for constants by default. 
If you change the name of the property, you will receive a warning `Const property name 'DisabledChipContainerAlpha' should not contain lowercase letters`, and if you run `spotlessApply`, it will fail with the message `rule: standard:property-naming, Property name should use the screaming snake case notation when the value can not be changed`

https://pinterest.github.io/ktlint/1.0.0/rules/standard/#property-naming

We can discuss which style we should use in the project, and add customized rules to allow it. However, we should stick with the default rule for now since ktlint is our current code formatter.

**Reply by diskerr** [reply]
Thanks. I got it. It's the default rule of ktlint. Hopefully, one day the rule will be modified to use a single consistent style. I wish that it would adopt the Compose style.
